dataclients/kubernetes/definitions.go

```diff
@@ -174,20 +193,52 @@ type endpointList struct {
 	Items []*endpoint `json:"items"`
 }
 
-func (ep endpoint) Targets(svcPortName, svcPortTarget string) []string {
+func formatEndpoint(a *address, p *port) string {
+	return fmt.Sprintf("http://%s:%d", a.IP, p.Port)
+}
+
+func (ep endpoint) targets(svcPortName, svcPortTarget string) []string {
 	result := make([]string, 0)
 	for _, s := range ep.Subsets {
 		for _, port := range s.Ports {
+			// TODO: verify if the comparison of port.Name == svcPortName is valid,
+			// considering that the svcPortName is the name of service port specified
+			// in ingress. The right way probably is not to use the svcPortName here,
+			// since that's a reference from the ingress backend to the service, but
+			// use the svcPortTarget, which can be a name or a number, to compare it
+			// with the subset port.Name and port.Port, which is referenced by the
+			// service target port, which can also be either a name or a port value.
```

Comment:
`port.Name == svcPortName` is valid.

Reply:
related to other comments and this branch: https://github.com/zalando/skipper/tree/improvement/service-port-fallback-handling